das/coordinator_test.go

```diff
@@ -72,11 +78,14 @@ func TestCoordinator(t *testing.T) {
 	})
 
 	t.Run("prioritize newly discovered over known", func(t *testing.T) {
+
+		// params := dasingParams{}
+		params.concurrencyLimit = 1
+		params.samplingRange = 4
+
```

Comment:
The issue here that previously in tests variables were shadowed inside each test if they had to be changed, meaning no change of the original variable. But now you modify value of `Params` struct  inside test so it affects all tests that depend on it. In case you need to modify config, easy fix could be to make a copy of config first and then modify the copy. Looks like you had an idea of it by introducing shadowing with copy, but commented it out.
Example of solution:
```
t.Run("priority routine should not lock other workers", func(t *testing.T) {
  tParams := params
  tParams. concurrencyLimit = 100
  tParams.samplingRange = 4
  
  .. use tParams 
}
```
```